conan/tools/meson/toolchain.py

```diff
@@ -71,7 +73,9 @@ def __init__(self, conanfile, env=os.environ):
         self._shared = self._conanfile.options.get_safe("shared")
         self._fpic = self._conanfile.options.get_safe("fPIC")
         self.definitions = dict()
-        self._env = env
+        self.preprocessor_definitions = dict()
+        self._env = env.copy()
```

Comment:
Leave the ``env`` outside of this PR. This PR can implement the ``preprocessor_definitions`` without env

Reply:
no, unfortunately we cannot.
if I override `cflags` with preprocessor definitions, it no longer uses default and no longer fetches `CFLAGS` from the environment.

Reply:
Isn't a way in Meson in which you can just define the preprocessor definitions, and append them to existing in the Meson toolchain (not in Conan)? BTW, just t make sure, these are not CFLAGS at all, these would be CPPFLAGS, right?

Reply:
I didn't find any way to just add preprocessor definitions on their own. the only way seems to be apply them to `c_args`.

Reply:
It seems we will need to wait until we correctly propagate the environment, unless a different way that doesn't require reading the CFLAGS env-var is necessary.

Reply:
okay, we can wait. no-one actually requested it (yet). I've added just for the feature parity between toolchains.